### ISSUE TITLE
When trying to reproduce results for QQP the train classifier always predicts one of the two classes.

### ISSUE THREAD

**Issue by bvasilis** [google-research/bert#475]
Hello,

I'm trying to reproduce the results for QQP and, after training the model for a few epochs (2, 3, or 4), the model consistently predicts that all the test set examples pertain to only one of the two classes. The dev set accuracy is however normal (87%). What's even weirder is that whenever I'm re-running the prediction step, the model sometimes predicts that all test examples are True, and other times that they are all false.

I'm attaching the preprocessing code and commands I'm using in case the fault is there:

QQP Preprocessing:
```python
 class QuoraProcessor(DataProcessor):
     """Processor for Quora Question duplicate dataset (GLUE Version)"""
     def get_train_examples(self, data_dir):
       return self._create_examples(
         self._read_tsv(os.path.join(data_dir, "train.tsv")), "train" )
 
     def get_dev_examples(self, data_dir):
       return self._create_examples(
         self._read_tsv(os.path.join(data_dir, "dev.tsv")) , "dev" )
 
     def get_test_examples(self, data_dir):
       return self._create_examples(
         self._read_tsv(os.path.join(data_dir, "test.tsv")), "test" )
 
     def get_labels(self):
       return ["0", "1"]
 
     def _create_examples(self, lines, set_type):
      examples = []
      for (i, line) in enumerate(lines):
        if i == 0:
          continue
        guid = "%s - %s" % (set_type, i)
        if set_type == "test":
          question_a = tokenization.convert_to_unicode(line[1])
          question_b = tokenization.convert_to_unicode(line[2])
          label = "0"
        else:
          if not len(line)==6:
            continue
          question_a = tokenization.convert_to_unicode(line[3])
          question_b = tokenization.convert_to_unicode(line[4])
          label = tokenization.convert_to_unicode(line[5])
        examples.append(
          InputExample(guid=guid, text_a=question_a, text_b=question_b, label=label))
      return examples
```

Training command:
```bash
python run_classifier.py   --task_name=QQP   --do_train=true   --do_eval=true   --do_lower_case=True   --data_dir=$GLUE_DIR/QQP   --vocab_file=$BERT_BASE_DIR/vocab.txt   --bert_config_file=$BERT_BASE_DIR/bert_config.json   --init_checkpoint=$BERT_BASE_DIR/bert_model.ckpt   --max_seq_length=128   --train_batch_size=24   --learning_rate=2e-5   --num_train_epochs=4.0   --output_dir=./outputs
```

Prediction command:
```bash
python run_classifier.py   --task_name=QQP   --do_predict=true   --data_dir=$GLUE_DIR/QQP   --vocab_file=$BERT_BASE_DIR/vocab.txt   --bert_config_file=$BERT_BASE_DIR/bert_config.json   --init_checkpoint=./outputs/model.ckpt-60641.index --do_lower_case=True  --max_seq_length=128 --output_dir=./checkpoints
```

Would love some help here, I'm stuck for some time now. Thanks!

**Comment by sc89703312** [comment]
Did u solve this problem?

**Comment by sc89703312** [comment]
Have u calculate the precision, recall and F1 on eval set, maybe only accuracy can't stand for model performance

**Comment by wangjs9** [comment]
Same problem..
